### PR TITLE
Add SNPS RMX500 platform support in terms of HWMv2

### PR DESCRIPTION
**Pull request by kokas-a**
Draft PR to make internal discussion about the concept itself in general

### CODE REVIEW

File: boards/snps/nsim/arc/Kconfig
Comment (nordicjm) [inline boards/snps/nsim/arc/Kconfig:1]: do not rename folder to have arc in them, we are not going back to hwmv1

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Comment by kokas-a** [comment]
@evgeniy-paltsev, @ruuddw , @abrodkin , please join to discussion

File: boards/snps/nsim/arc/Kconfig
Comment (kokas-a) [reply boards/snps/nsim/arc/Kconfig:1]: Hi @nordicjm,

Such scheme of naming was suggested in internal Synopsys discussion. The point is board name "nsim" - it is emulator, that supports 2 architectures, and we need somehow separate variants of the same "board" with different cores inside (arc and risc-v). Could you adwise the correct way to to that?

Comment (nordicjm) [reply boards/snps/nsim/arc/Kconfig:1]: By naming the files with the CPU cluster in it, for example https://github.com/zephyrproject-rtos/zephyr/tree/main/boards/nxp/imx8mp_evk which has an ARM core (cortex M4, CPU cluster is `m4`), AARCH64 (ARM64) core (cortex a53, CPU cluster is `a53`) and xtensa core (tensilica HiFi4, CPU cluster is `adsp`)

Comment (ruuddw) [reply boards/snps/nsim/arc/Kconfig:1]: This is different, nsim is an instruction set simulator. The pattern followed here is like qemu. It has various simulated CPU flavors grouped in subdirs. nsim doesn't have all CPUs together in one SoC, these are "SoC" variants with a different CPU.

To make this more clear, I'd recomment to rename arc to arcv2 (not fully accurate, as it has some arcv3 cores in there but close enough, arc_v2_3 and arc_v5 may be better?).

Comment (kokas-a) [reply boards/snps/nsim/arc/Kconfig:1]: Yes, indeed. Qemu is good exanple to describe situatoin with nSIM

**Comment by kokas-a** [comment]
Replaced with new one: https://github.com/zephyrproject-rtos/zephyr/pull/72021

**Pull request closed without merging**